services/streams/streams/parsers.go

```diff
@@ -210,15 +216,19 @@ func (streamComp *StreamComponent) FromMap(streamCompMap map[string]any) error {
 		streamComp.Endianness = streamCompMap["endianness"].(string)
 	}
 	if streamComp.Type == "SWITCH" {
-		if err := streamComp.Expression.FromMap(streamCompMap["expression"].([]any)[0].(map[string]any)); err != nil {
-			return err
+		if len(streamCompMap["expression"].([]any)) > 0 {
+			if err := streamComp.Expression.FromMap(streamCompMap["expression"].([]any)[0].(map[string]any)); err != nil {
+				return err
+			}
 		}
 	}
 	if streamComp.Type == "SWITCH" || streamComp.Type == "CONTAINER" {
-		if elements, err := helper.ParseFromMaps[StreamComponent](streamCompMap["elements"].([]any)); err != nil {
-			return err
-		} else {
-			streamComp.Elements = elements
+		if len(streamCompMap["elements"].([]any)) > 0 {
+			if elements, err := helper.ParseFromMaps[StreamComponent](streamCompMap["elements"].([]any)); err != nil {
+				return err
+			} else {
+				streamComp.Elements = elements
```

Comment:
can you just set this instead of having the else?
Same for other places in this method
Sonar just complained about the Cognitive Complexity which is too high